nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -375,13 +482,16 @@ in {
         { assertion = !(acfg.dbpass != null && acfg.dbpassFile != null);
           message = "Please specify no more than one of dbpass or dbpassFile";
         }
-        { assertion = ((acfg.adminpass != null || acfg.adminpassFile != null)
-            && !(acfg.adminpass != null && acfg.adminpassFile != null));
+        { assertion = (acfg.adminpass != null) != (acfg.adminpassFile != null);
```

Comment:
Please do this in a separate commit. While the change itself seems correct, I'd prefer to always do it. That way maintainers don't get commits when bisecting for an issue with a lot of unrelated changes.